exonum/src/storage/proof_map_index/mod.rs

```diff
@@ -352,84 +292,47 @@ where
-                        self.construct_proof(&branch, &suf_searched_path);
-
-                    if let Some(child_proof) = proof_from_level_below {
-                        let child_proof_pos = suf_searched_path.bit(0);
-                        let neighbor_child_hash = *branch.child_hash(!child_proof_pos);
-                        match child_proof_pos {
-                            ChildKind::Left => MapProof::Branch(BranchProofNode::LeftBranch {
-                                left_node: Box::new(child_proof),
-                                right_hash: neighbor_child_hash,
-                                left_key: l_s,
-                                right_key: r_s,
-                            }),
-                            ChildKind::Right => MapProof::Branch(BranchProofNode::RightBranch {
-                                left_hash: neighbor_child_hash,
-                                right_node: Box::new(child_proof),
-                                left_key: l_s,
-                                right_key: r_s,
-                            }),
-                        }
-                    } else {
-                        let l_h = *branch.child_hash(ChildKind::Left); //copy
-                        let r_h = *branch.child_hash(ChildKind::Right); //copy
-                        MapProof::Branch(BranchProofNode::BranchKeyNotFound {
-                            left_hash: l_h,
-                            right_hash: r_h,
-                            left_key: l_s,
-                            right_key: r_s,
-                        })
-                        // proof of exclusion of a key, because none of child paths is a
-                        // prefix(searched_path)
-                    }
-                } else {
-                    // if common prefix length with root_path is less than root_path length
-                    let l_h = *branch.child_hash(ChildKind::Left); //copy
-                    let r_h = *branch.child_hash(ChildKind::Right); //copy
-                    MapProof::Branch(BranchProofNode::BranchKeyNotFound {
-                        left_hash: l_h,
-                        right_hash: r_h,
-                        left_key: l_s,
-                        right_key: r_s,
-                    })
-                    // proof of exclusion of a key, because root_path != prefix(searched_path)
-                }
-            }
-            None => MapProof::Empty,
-        }
+    /// Returns the combined proof of existence or non-existence for the multiple specified keys.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use exonum::storage::{MemoryDB, Database, ProofMapIndex};
+    ///
+    /// let db = MemoryDB::new();
+    /// let snapshot = db.snapshot();
+    /// let index: ProofMapIndex<_, [u8; 32], u8> = ProofMapIndex::new("index", &snapshot);
+    ///
+    /// let proof = index.get_multiproof(vec![[0; 32], [1; 32]]);
+    /// # drop(proof);
```

Comment:
For what?

Reply:
`drop` is used in some existing `ProofMapIndex` code samples, so I just copied it from there. Do you propose to remove it?

Reply:
We have such `drop`s in other tests. As far as I can remember, they were added to suppress the compiler warnings in one specific case: https://github.com/intellij-rust/intellij-rust/issues/1554
I suppose it can be removed.